internal/auth0/organization/resource_members.go

```diff
@@ -0,0 +1,203 @@
+package organization
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+	"github.com/auth0/terraform-provider-auth0/internal/value"
+)
+
+// NewMembersResource will return a new auth0_organization_members (1:many) resource.
+func NewMembersResource() *schema.Resource {
+	return &schema.Resource{
+		Description:   "This resource is used to manage members of an organization.",
+		CreateContext: createOrganizationMembers,
+		ReadContext:   readOrganizationMembers,
+		UpdateContext: updateOrganizationMembers,
+		DeleteContext: deleteOrganizationMembers,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"organization_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the organization to assign the member to.",
+			},
+			"members": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required:    true,
+				Description: "Add user ID(s) directly from the tenant to become members of the organization.",
+			},
+		},
+	}
+}
+
+func createOrganizationMembers(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+
+	organizationID := data.Get("organization_id").(string)
+
+	mutex.Lock(organizationID)
+	defer mutex.Unlock(organizationID)
+
+	alreadyMembers, err := api.Organization.Members(organizationID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	data.SetId(organizationID)
```

Comment:
Suggestion - perhaps set the ID after successfully assigning members to the org. That way you can remove the `data.SetId("")` on line 76.

Reply:
While that might be nicer and tidier code wise, if the organization exists we should immediately set the ID of this resource IMO. Any error resulting from the `AddMembers` call shouldn't erase the resource from the state (that's what happens if we don't set the ID).